sdk/core/azure-core-http-httpurlconnection/src/main/java/com/azure/core/http/httpurlconnection/HttpUrlConnectionClient.java

```diff
@@ -0,0 +1,152 @@
+package com.azure.core.http.httpurlconnection;
+
+import com.azure.core.http.*;
+import com.azure.core.http.httpurlconnection.implementation.HttpUrlConnectionResponse;
+import com.azure.core.util.Context;
+import reactor.core.publisher.Mono;
+
+import java.io.*;
+import java.net.*;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * This class provides a HttpUrlConnection implementation for the {@link HttpClient} interface.
+ * Creating an instance of this class can be achieved by using the {@link HttpUrlConnectionClientBuilder} class.
+ *
+ * @see HttpClient
+ * @see HttpUrlConnectionClientBuilder
+ */
+public class HttpUrlConnectionClient implements HttpClient {
+
+    // Asynchronous send method returning a Mono of HttpResponse
+    @Override
+    public Mono<HttpResponse> send(HttpRequest httpRequest) {
+        return sendAsync(httpRequest, Context.NONE);
+    }
+
+    // Asynchronous send method returning a Mono of HttpResponse
+    @Override
+    public Mono<HttpResponse> send(HttpRequest request, Context context) {
+        return sendAsync(request, context);
+    }
+
+    // Synchronous send method with additional context, primarily for interface compliance
+    @Override
+    public HttpResponse sendSync(HttpRequest httpRequest, Context context) {
+        return sendAsync(httpRequest, context).block();
+    }
+
+    /**
+     * Asyncronously send the HttpRequest
+     *
+     * @param httpRequest The HTTP Request being sent
+     * @param context The context of the request, for any additional changes
+     * @return A Mono containing a HttpResponse object
+     */
+    public Mono<HttpResponse> sendAsync(HttpRequest httpRequest, Context context) {
+        HttpMethod httpMethod = httpRequest.getHttpMethod();
+        if (httpMethod == HttpMethod.PATCH) {
+            return sendPatchViaSocket(httpRequest);
+        }
+        return openConnection(httpRequest)
+            .flatMap(connection -> setConnectionRequest(connection, httpRequest)
+                .then(writeRequestBody(connection, httpRequest))
+                .then(readResponse(connection, httpRequest))
+                .doFinally(signalType -> connection.disconnect()) // Disconnect connection after processing
+                .onErrorResume(e -> Mono.error(new RuntimeException(e))));
+    }
+
+    // Send a PATCH request via a SocketClient
+    // Send a PATCH request via a SocketClient
+    private Mono<HttpResponse> sendPatchViaSocket(HttpRequest httpRequest) {
+        return Mono.fromCallable(() -> {
+            try {
+                return new SocketClient(httpRequest.getUrl()).sendPatchRequest(httpRequest);
```

Comment:
One of the comments I raised previously was - is this a good idea to new up a new client for every request?